prql-compiler/src/semantic/lowering.rs

```diff
@@ -195,7 +195,7 @@ impl Lowerer {
             ast::TransformKind::Aggregate { assigns, tbl } => {
                 let select = self.declare_as_columns(assigns, &mut transforms)?;
 
-                transforms.push(Transform::Aggregate(select));
+                transforms.push(Transform::Aggregate { by: select });
                 Some(tbl)
```

Comment:
`by` should now contain `transform_call.partition` instead of assings (select columns).

Assigns should go into a `Transfrom::Select` that is pushed into transforms *before* `Aggregate`.

I should have written more about this in the issue.